ska_conda/ska_builder.py

```diff
@@ -11,32 +11,54 @@ class SkaBuilder(object):
 
     def __init__(self, ska_root=None):
         if ska_root is None:
-            ska_root = "/data/acis/ska3_pkg/"
+            ska_root = "/tmp/ska3_pkg/"
         self.ska_root = ska_root
         self.ska_build_dir = os.path.join(self.ska_root, "builds")
         self.ska_src_dir = os.path.join(self.ska_root, "src")
         os.environ["SKA_TOP_SRC_DIR"] = self.ska_src_dir
 
-    def _clone_repo(self, name):
+    def _clone_repo(self, name, tag=None):
         if name == "ska":
             return
         print("Cloning source %s." % name)
         clone_path = os.path.join(self.ska_src_dir, name)
-        if os.path.exists(clone_path):
-            print("Source %s exists, skipping." % name)
-            return
-        yml = os.path.join(pkg_defs_path, name, "meta.yaml")
-        with open(yml) as f:
-            requires = False
-            while not requires:
-                line = f.readline().strip()
-                if line.startswith("path:"):
-                    requires = True
-            data = yaml.load(f)
-            url = data['about']['home']
-            git.Repo.clone_from(url, clone_path)
+        if not os.path.exists(clone_path):
+            # Try ssh first to avoid needing passwords for the private repos
+            # We could add these ssh strings to the meta.yaml for convenience
+            try:
+                git_ssh_path = 'git@github.com:sot/' + name + '.git'
+                repo = git.Repo.clone_from(git_ssh_path, clone_path)
+                assert not repo.bare
+            except:
+                yml = os.path.join(pkg_defs_path, name, "meta.yaml")
+                with open(yml) as f:
+                    requires = False
+                    while not requires:
+                        line = f.readline().strip()
+                        if line.startswith("path:"):
+                            requires = True
+                    data = yaml.load(f)
+                    url = data['about']['home']
+                repo = git.Repo.clone_from(url, clone_path)
+        else:
+            repo = git.Repo(clone_path)
```

Comment:
Good point!  Though I'm wondering if we need to think about the use cases and problems of  the "already-existing" dir some more anyway .

Reply:
We stipulate that the existing directory is created and maintained only via this script.  Problems then seem extremely unlikely.

Reply:
OK, so do we need a way to build a custom/test package?  Or would we just do that outside this tool?

Reply:
I think we should be able to handle this procedurally by making a release candidate tag such as `3.43.4rc1`.  Apart from testing the packaging itself, one can do functional/integration testing using pip install from the development git repo.  I.e. spin up a dev Ska3, conda uninstall the package in question, pip install the dev version, then test.  Or conda install using the local build path.  Since "conda acknowledges PEP440", this *should* work.

But let's defer further specifics of this use case for a future PR if needed, it will be easier based on an already-working system.  I'll stub a placeholder into the process wiki.

Reply:
So adding `get fetch origin` seems fine if you want to reuse the repos dirs over some period of time and want to make sure you have the newest tags at origin.  We could also just clone the repos fresh for every run of the tool and remove this code path, but that does seem a little annoying.

Reply:
Yes, I envision an "official" build process for production, done as `aca` user, that uses a permanent location like `/proj/sot/ska3/conda-builds`, or something like that.  Fetching and pulling is definitely faster than re-cloning the whole thing every time.